src/modules/NetworkController/private/Update-NetworkControllerGlobalConfig.ps1

```diff
@@ -0,0 +1,140 @@
+                finally {
+                    if ($null -ne $client) {
+                        $client.Dispose()
+                    }
+                }
+            }
+
+            Write-Host "Updating NetworkController Global Config with certs"
+            $NcNodeCertInfo
+
+            Write-Host "Current NetworkController Global Config:"
+            $newConfigs = [System.Collections.ArrayList]::new()
+
+            Connect-ServiceFabricCluster | Out-Null
+            $client = [System.Fabric.FabricClient]::new()
+            $result = $null
+            $method = [System.Fabric.NamedProperty].getmethod("GetValue").MakeGenericMethod([string])
+            do {
+                $result = $client.PropertyManager.EnumeratePropertiesAsync("fabric:/NetworkController/GlobalConfiguration", $true, $result).Result
+                $result.GetEnumerator() | ForEach-Object {
+                    $name = $_.Metadata.PropertyName
+                    $value = $method.Invoke($_, $null);
+                    if ($name.StartsWith("Global") -and ($name.Contains("Thumbprint") -or $name.Contains("SSL"))) {
+                        "Name: " + $name + " , " + "Value: " + $value
+
+                        $newConfig = $null
+                        if ($name.Contains("SSL")) {
+                            # $newConfig = [PSCustomObject]@{
+                            #     Name  = $name
+                            #     Value = $using:ncNodeCertInfo['RestCert']
+                            # }
+                        }
+                        elseif ($name.Contains("Thumbprint")) {
+                            foreach ($certInfoKey in $NcNodeCertInfo.Keys) {
+                                Write-Verbose "Looking for node match $name"
+                                if ($name -match $certInfoKey) {
+                                    $newConfig = [PSCustomObject]@{
+                                        Name  = $name
+                                        Value = $($NcNodeCertInfo).$certInfoKey
+                                    }
+                                    Write-Verbose "Found thumbprint: $($newConfig.Value)"
+                                }
+                            }
+                        }
+
+                        if ($null -ne $newConfig) {
+                            $newConfigs.Add($newConfig) | Out-Null
+                        }
+                    }
+                }
+            }
+            while ($result.HasMoreData)
+
+            Write-Host "Updating with new NetworkController Global Config:"
+        
+            foreach ($newConfig in $newConfigs) {
+                Write-Host "name: $($newConfig.Name) value: $($newConfig.Value)"
+                PutProperty -Property $($newConfig.Name) -Value $($newconfig.Value)
+            }
+        }
+
+        $NodeFQDN = (get-ciminstance win32_computersystem).DNSHostName + "." + (get-ciminstance win32_computersystem).Domain
+        if($NcVMs -contains $NodeFQDN){
```

Comment:
`Test-ComputerNameIsLocal -ComputerName $NcVms[0]` would help determine if local vs remote invocation should happen

Reply:
Refactored